katas/012_vector_search.md

```diff
@@ -0,0 +1,212 @@
+# Twelfth Kata: Vector Search
+
+
+## Understanding Vector Search
+
+Let's dive into Vector Search!
+
+What we refer to as "vector search" is also known as "neural search", or "dense vector search", or sometimes "cognitive search", among other names.
+There are two types of vector representation : Sparse Vectors and Dense Vectors.
+
+In traditional search, we use inverted index for searching through the content of documents. We can understand an inverted index as a derivative of a sparse vector model of N dimensions, where N is the size of the vocabulary (after stemming, removing stopwords etc.).
+Each dimension/axis represents a token in the index and each document is represented by an embedding or a vector of a 1 or a non-zero whole number (e.g. term frequency) for the token present in the given document, or 0 when missing. 
```

Comment:
We could link https://opensourceconnections.com/blog/2019/12/18/bert-and-search-relevance-part2-dense-vs-sparse/ here. In my understanding you would see each document as a vector of N dimensions with N being the number of terms in the index. The values could be 0/1 (term not/present) or tf*idf. I wouldn't use the words `derivative` and `embedding`.